content/blog/clickhouse-benchmarking.md

```diff
@@ -0,0 +1,188 @@
+---
+title: "ClickHouse Performance Benchmarking on ECS"
+authorId: "rohit"
+date: 2024-10-21
+draft: false
+featured: true
+weight: 1
+---
+
+Imagine being a Formula One driver, racing at breakneck speeds, but without any telemetry data to guide you. It’s a thrilling ride, but one wrong turn or overheating engine could lead to disaster. Just like a pit crew relies on performance metrics to optimize the car's speed and handling, we use observability in ClickHouse to monitor our data system's health. These metrics provide crucial insights, allowing us to identify bottlenecks, prevent outages, and fine-tune performance, ensuring our data engine runs as smoothly and efficiently as a championship-winning race car.
+
+In this blog, we’ll focus on the performance benchmarking of ClickHouse on AWS ECS during the ingestion of different data volumes. We’ll analyze key system metrics such as CPU usage, memory consumption, disk I/O, and row insertion rates across varying data ingestion sizes.
+
+For setting up the ClickHouse cluster, we followed the [ClickHouse replication architecture guide](https://clickhouse.com/docs/en/architecture/replication) and the [AWS CloudFormation ClickHouse cluster setup](https://aws-ia.github.io/cfn-ps-clickhouse-cluster/). Using these resources, we replicated the setup on ECS, allowing us to run performance benchmarking tests on the environment.
+
+By examining performance metrics during the ingestion of 1 million (10 lakh), 5 million (50 lakh), 10 million (1 crore), and 66 million (6.6 crore) logs, we aim to provide a quantitative analysis of how system behavior changes as the load increases.
```

Comment:
Can we set up all the scenarios you're looking at in this blog upfront?

1. Ingestion
    1. Metrics you want to look at
2. Querying
    1. Metrics you want to look at
3. Effect of node count
    1. Metrics you want to look at

Reply:
Also, there is no need to convert millions to an Indian numbering system(ex: `1 million (10 lakh)`). I think people understand the international numbering system.

Reply:
the metrics collected for all the three scenarios are the same one from the default clickhouse dashboard and we have already mentioning them in the blog under each scenario , so i think mentioning them again in the blog at the start would be more of repetative stuff and will make the blog quite lengthy